pydomcfg/tests/bathymetry.py

```diff
@@ -155,6 +155,9 @@ def _calc_rmax(depth):
         Slope steepness value (units: None)
     """
 
+    # Replace land with zeros
+    depth = depth.where(depth > 0, 0)
```

Comment:
I would say yes, since I think we should write a function that put all land values to zero when interpolating the input bathymetry onto the model grid. So the model bathymetry will be positive defined by definition, with zeros for land (also I think NEMO will fail if this condition are not met, but not sure about this last point)

Reply:
Sorry, I've been dipping in and out of this today (so no continuity in my thought process). @oceandie re NEMO will fail, NEMO as far as I understand doesn't use bathymetry field, just scale factors and wet cells [i.e. if top_level is equal to zero it's land]. So in that sense we can define land values in bathymetry as we see fit. I'll look over the python again tomorrow (and you may have already included some of this), but the key user defined vars are `rn_sbot_min` (for `ln_sco`) and `rn_hmin` (for `ln_zco`, `ln_zps`; which take different meanings depending on whether they are -ve or +ve: min number of levels or min depth). So when calculating rmax this user choices will affect the outcome (although, not in the case of `tests`).

from the Fortran (where the bathy is updated):

```
IF ( .not. ln_sco ) THEN                                !==  set a minimum depth  ==!
         IF( rn_hmin < 0._wp ) THEN    ;   ik = - INT( rn_hmin )                                      ! from a nb of level
         ELSE                          ;   ik = MINLOC( gdepw_1d, mask = gdepw_1d > rn_hmin, dim = 1 )  ! from a depth
         ENDIF
         zhmin = gdepw_1d(ik+1)                                                        ! minimum depth = ik+1 w-levels 
         WHERE( bathy(:,:) <= 0._wp )   ;   bathy(:,:) = 0._wp                         ! min=0     over the lands
         ELSE WHERE ( risfdep == 0._wp );   bathy(:,:) = MAX(  zhmin , bathy(:,:)  )   ! min=zhmin over the oceans
         END WHERE
         IF(lwp) write(numout,*) 'Minimum ocean depth: ', zhmin, ' minimum number of ocean levels : ', ik
ENDIF
```

and

```
bathy(:,:) = MIN( rn_sbot_max, bathy(:,:) )

         DO jj = 1, jpj
            DO ji = 1, jpi
              IF( bathy(ji,jj) > 0._wp )   bathy(ji,jj) = MAX( rn_sbot_min, bathy(ji,jj) )
            END DO
         END DO
```

note there is also a `rn_sbot_max`

Apologies if I'm going over old ground...

Reply:
HI @jdha, yes sorry, I was still thinking about NEMO3.6 or DOMCFG, that we are actually trying to replicate :) 
Regarding all the params and bits of code you mentioned, yes I started to consider them - soon I will push where I am now in sco_dev , then any feedback is more than welcome ;)

Reply:
OK, this is helpful! (maybe open an issue about this as this comment will get kind of lost when we merge this PR?)

This PR should be just the starting point, and `_calc_rmax` will become the general function to compute `rmax`. Once we merge this PR, I think @oceandie will merge main into #33 and will move the whole `_calc_rmax` function into `utils`. I guess vcoord-specific parameters such as `rn_sbot_min` will be handled internally by our domzgr classes (but all classes will make use of the same `_calc_rmax` function).

Reply:
Hi @malmans2  and @jdha , how can I see the original code for calc_rmax by @jdha (the one using numpy) ?

Reply:
This PR started from the xarray version. You'd have to look in James' PR (already merged). Should be this one: https://github.com/pyNEMO/pyDOMCFG/pull/17

I'd go there and look at the appropriate commit (e.g., click on commits and select the first commit).

Reply:
Thanks!